src/conn/conn_api.c

```diff
@@ -1285,6 +1285,35 @@ __conn_rollback_to_stable(WT_CONNECTION *wt_conn, const char *config)
 err:	API_END_RET(session, ret);
 }
 
+/*
+ * __conn_calibrate_ticks --
+ *	Calibrate a ratio from rdtsc ticks to nanoseconds.
+ */
+static void
+__conn_calibrate_ticks(WT_SESSION_IMPL *session)
+{
+#if (defined __i386) || (defined __amd64)
+	struct timespec start, stop;
+	uint64_t diff_nsec, diff_tsc;
+	uint64_t i, tsc_start, tsc_stop;
+
+	__wt_epoch(session, &start);
+	tsc_start = __wt_rdtsc(session);
+	/*
+	 * This needs to be CPU intensive and large enough.
```

Comment:
Is the loop for a million operations enough? I suspect the compiler would just remove that entirely unless we do something more involved. It's neat to have this in a single function call, but maybe we could wrap it around some real work done by `wiredtiger_open`? Maybe starting https://github.com/wiredtiger/wiredtiger/blob/develop/src/conn/conn_api.c#L2352 and ending around https://github.com/wiredtiger/wiredtiger/blob/develop/src/conn/conn_api.c#L2599

Reply:
My investigation in the ticket indicates that we really want something self-contained and clearly CPU intensive, with no option of I/O or other interference that can lead to inconsistent measurements. A million seemed sufficient in my experiments giving a large enough number of ticks to compute the ratio.

Reply:
You are correct, good catch. If I declare `i` as `volatile` however it works as expected and that loop does not get optimized away.  I tested various values for the loop, from 1M to 1B and they all calculated the same ratio out to 4 decimal places. But there was a noticeable delay with 1B.  I think for our level of accuracy, 1M will be sufficient.